File: src/main/java/com/hanghae/naegahama/repository/UserRepository.java
package com.hanghae.naegahama.repository;

import com.hanghae.naegahama.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByUserEmail(String userEmail);
}

File: src/main/kotlin/com/example/demozeroset/repository/UserRepository.kt
package com.example.demozeroset.repository

import com.example.demozeroset.domain.User
import org.bson.types.ObjectId
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.querydsl.QuerydslPredicateExecutor
import org.springframework.stereotype.Repository

@Repository
interface UserRepository: MongoRepository<User, ObjectId> , QuerydslPredicateExecutor<User>{
}

